client/data/settings/hooks.js

```diff
@@ -31,17 +81,59 @@ export const useSettings = () => {
 	return { settings, isLoading, isSaving, saveSettings };
 };
 
-export const usePaymentRequestEnabledSettings = () => {
-	const { updateIsPaymentRequestEnabled } = useDispatch( STORE_NAME );
-
-	const isPaymentRequestEnabled = useSelect( ( select ) => {
-		const { getSettings } = select( STORE_NAME );
-
-		return getSettings().is_payment_request_enabled || false;
-	}, [] );
-
-	return [ isPaymentRequestEnabled, updateIsPaymentRequestEnabled ];
-};
+export const useEnabledPaymentMethodIds = makeSettingsHook(
+	'enabled_payment_method_ids',
+	EMPTY_ARR
+);
+export const usePaymentRequestEnabledSettings = makeSettingsHook(
+	'is_payment_request_enabled'
+);
+export const usePaymentRequestButtonSize = makeSettingsHook(
+	'payment_request_button_size',
+	''
+);
+export const usePaymentRequestButtonType = makeSettingsHook(
+	'payment_request_button_type',
+	''
+);
+export const usePaymentRequestButtonTheme = makeSettingsHook(
+	'payment_request_button_theme',
+	''
+);
+export const useIsStripeEnabled = makeSettingsHook( 'is_stripe_enabled' );
+export const useTestMode = makeSettingsHook( 'is_test_mode_enabled' );
+export const useSavedCards = makeSettingsHook( 'is_saved_cards_enabled' );
+export const useManualCapture = makeSettingsHook( 'is_manual_capture_enabled' );
+export const useSeparateCardForm = makeSettingsHook(
+	'is_separate_card_form_enabled'
+);
+export const useAccountStatementDescriptor = makeSettingsHook(
+	'statement_descriptor',
+	''
+);
+export const useIsShortAccountStatementEnabled = makeSettingsHook(
+	'is_short_statement_descriptor_enabled'
+);
+export const useShortAccountStatementDescriptor = makeSettingsHook(
+	'short_statement_descriptor',
+	''
+);
+export const useDebugLog = makeSettingsHook( 'is_debug_log_enabled' );
+
+export const usePaymentRequestLocations = makeSettingsHookFromUpdateHandler(
+	'payment_request_button_locations',
+	( v, updateSettingsValues ) =>
+		updateSettingsValues( {
+			payment_request_button_locations: [ ...v ],
+		} ),
+	EMPTY_ARR
+);
```

Comment:
Why can't this use `makeSettingsHook` directly? The value is still being updated in the redux store. And the array reference is already changing in the `PaymentRequestSection` component.

If we used `makeSettingsHook`, we could get away with just having a `makeReadOnlySettingsHook` (for read only) and `makeSettingsHook`, without the need of a third "factory" function